src/chainstate/stacks/miner.rs

```diff
@@ -56,6 +56,149 @@ use util::vrf::*;
 use core::*;
 use core::mempool::*;
 
+///
+///    Independent structure for building microblocks:
+///       StacksBlockBuilder cannot be used, since microblocks should only be broadcasted
+///       once the anchored block is mined, won sortition, and a StacksBlockBuilder will
+///       not survive that long.
+///
+///     StacksMicroblockBuilder holds a mutable reference to the provided chainstate in the
+///       new function. This is required for the `clarity_tx` -- basically, to append transactions
+///       as new microblocks, the builder _needs_ to be able to keep the current clarity_tx "open" 
+pub struct StacksMicroblockBuilder<'a> {
+    anchor_block: BlockHeaderHash,
+    anchor_block_bhh: BurnchainHeaderHash,
+    anchor_block_height: u64,
+    prev_microblock_header: Option<StacksMicroblockHeader>,
+    header_reader: StacksChainState,
+    clarity_tx: Option<ClarityTx<'a>>,
+    considered: Option<HashSet<Txid>>,
```

Comment:
We will need to track the total epoch's size in this struct.

Reply:
Updated in 808abde